examples/merge_sort.py

```diff
@@ -0,0 +1,95 @@
+"""Example showing merge sort using BoC."""
+
+import random
+from time import sleep
+
+from pyrona import Region, wait, when
+
+
+Threshold = 10
+
+
+def _sort_section(source: tuple, start: int, end: int, output: Region):
+    if end < start:
+        return
+
+    if end - start + 1 <= Threshold:
+        print("sorting", start, end)
+        # below a threshold, we use the built-in sort function
+        values = list(sorted(source[start:end + 1]))
+
+        # when output:
+        @when(output)
```

Comment:
As every `when` uses the same `output` doesn't that mean there is no concurrency?